static/js/components/inputs/inputs_test.js

```diff
@@ -78,19 +81,46 @@ describe('Profile inputs', () => {
 
     it('should update the selected option when onChange fires with an option', () => {
       selectField = renderGenderSelectField().find(VirtualizedSelect);
-      let option = genderOptions[1];
-      selectField.props().onChange({
-        text: option.label,
-        value: option.value
-      });
-      assert.equal(inputProps.profile.gender, 'f');
+      modifyWrapperSelectField(selectField, 'f');
+      assert.ok(updateProfileStub.calledWith({
+        account_privacy: 'private',
+        first_name: '',
+        date_of_birth: '',
+        gender: 'f',
+        date_field: ''
+      }));
     });
 
     it('should have the correct option selected', () => {
       inputProps.profile.gender = 'f';
       selectField = renderGenderSelectField().find(VirtualizedSelect);
       assert.equal(selectField.props().value, 'f');
     });
+
+    it('should let you enter a new option, if the allowCreate option is passed', () => {
+      selectField = renderGenderSelectField({allowCreate: true}).find(VirtualizedSelect);
+      modifyWrapperSelectField(selectField, 'genderqueer');
+      assert.ok(updateProfileStub.calledWith({
+        account_privacy: 'private',
+        first_name: '',
+        date_of_birth: '',
+        gender: 'genderqueer',
+        date_field: ''
+      }), 'should be called with the right thing');
+    });
+
+    it('should enter the new option in the options list', () => {
+      selectField = renderGenderSelectField({allowCreate: true}).find(VirtualizedSelect);
+      modifyWrapperSelectField(selectField, 'genderqueer');
+      let newOption = selectField.props().options.find(option => (
```

Comment:
On this line, the id prop using `formattedId()`: https://github.com/mitodl/micromasters/pull/1831/files#diff-f57bc642ed5426be1405cd5b2aaca97dR125